routes/controllers/web-api.js

```diff
@@ -1,23 +1,41 @@
 
+const {pick} = require('lodash');
+
 async function find(req, res) {
-    const {
-        query,
-        property,
-        author,
-        sort,
-        start,
-        amount,
-    } = req.query;
-
-    if (!query && !author && !sort) {
+    const properties = {
+        query: req.query.query,
+        property: req.query.property,
+        author: req.query.author,
+        sort: req.query.sort,
+        start: req.query.start,
+        amount: req.query.amount,
+    };
+
+    // Remove undefined properties
+    Object.keys(properties).forEach((key) => {
+        if (properties[key] === undefined) delete properties[key];
+    });
+
+    // Put a singular `property` into an array
+    if (properties.property !== undefined && !Array.isArray(properties.property)) {
+        properties.property = [properties.property];
+    }
+
+    // Reject invalid case
+    if (
+        properties.query === undefined &&
```

Comment:
What if all of these are empty strings? Also, aren't we filtering out undefined properties above?